nucliadb/src/nucliadb/common/cluster/base.py

```diff
@@ -108,6 +108,21 @@ async def new_shard(
         resp = await self.writer.NewShard(req)  # type: ignore
         return resp
 
+    async def new_shard_with_vectorsets(
+        self,
+        kbid: str,
+        release_channel: utils_pb2.ReleaseChannel.ValueType,
+        vectorsets_configs: dict[str, VectorIndexConfig],
+    ) -> noderesources_pb2.ShardCreated:
+        req = NewShardRequest(
+            kbid=kbid,
+            release_channel=release_channel,
+            vectorsets_configs=vectorsets_configs,
+        )
+
+        resp = await self.writer.NewShard(req)  # type: ignore
```

Comment:
We should be validating that the response was successful no?
Or will the node return a grpc error in the event of an error?